src/eduid/workers/amapi/tests/test_user.py

```diff
@@ -0,0 +1,263 @@
+        super().setUp(am_users=[new_user_example])
+
+    def _make_url(self, endpoint: Optional[str] = None) -> str:
+        if endpoint is None:
+            return f"/users/{self.eppn}"
+        return f"/users/{self.eppn}/{endpoint}"
+
+    def _audit_log_tests(self, assert_diff: dict):
+        audit_log = self.api.audit_logger.get_by_eppn(self.eppn)
+        assert audit_log is not None
+        assert audit_log.eppn == self.eppn
+        assert audit_log.reason == self.reason
+        assert audit_log.source == self.source
+        assert audit_log.diff == self.as_json(assert_diff)
+
+    def make_put_call(self, req: dict, endpoint: Optional[str] = None):
+        response = self.client.put(
+            url=self._make_url(endpoint),
+            data=self.as_json(req),
+        )
+        assert response.status_code == 200
+
+    def make_delete_call(self, req: dict, endpoint: Optional[str] = None):
+        response = self.client.delete(
+            url=self._make_url(endpoint),
+            data=self.as_json(req),
+        )
+        assert response.status_code == 200
+
+    def test_update_name(self):
+        req = {
+            "reason": self.reason,
+            "source": self.source,
+            "given_name": None,
+            "display_name": "test_display_name",
+            "surname": "Smith",
+        }
+
+        assert_diff = {
+            "dictionary_item_removed": ["root['givenName']"],
+            "values_changed": {
+                "root['displayName']": {
+                    "new_value": "test_display_name",
+                    "old_value": "John Smith",
+                },
+            },
+        }
+
+        self.make_put_call(req, "name")
+
+        user_after = self.amdb.get_user_by_eppn(self.eppn)
+        assert user_after.given_name is None
+        assert user_after.display_name == "test_display_name"
+        assert user_after.surname == "Smith"
+        assert user_after.meta.version is not ObjectId("987654321098765432103210")
+
+        self._audit_log_tests(assert_diff=assert_diff)
+
+    def test_update_meta(self):
+        pass
+
+    def test_update_email(self):
+        req = {
+            "reason": self.reason,
+            "source": self.source,
+            "mail_addresses": [
+                {
+                    "email": "test@example.com",
+                    "created_by": "signup",
+                    "created_ts": "2013-09-02T10:23:25",
+                    "is_verified": True,
+                    "verified_by": "signup",
+                    "verified_ts": "2013-09-02T10:23:25",
+                    "is_primary": True,
+                    "modified_ts": "2013-09-02T10:23:25",
+                }
+            ],
+        }
+
+        assert_diff = {
+            "values_changed": {
+                "root['mailAliases'][0]": {
+                    "new_value": {
+                        "created_by": "signup",
+                        "created_ts": "2013-09-02T10:23:25",
```

Comment:
Vad är det som gör att vi tappar tidszonen här? Om svaret är "den finns inte där när vi läser in datat från mongodb" så undrar jag varför den verkar finnas där innan - är inte det också data läst från mongodb? :)